esy-lib/System.ml

```diff
@@ -112,4 +112,9 @@ module Environment = struct
     | Some path -> String.split_on_char sep.[0] path
     | None -> []
 
+  let homedir =
```

Comment:
Modeled after how node does it
https://github.com/nodejs/node/blob/883e1cd21d00a1ccf347c543c662b362383e86e0/lib/internal/modules/cjs/loader.js#L754
https://github.com/nodejs/node/blob/4e7d82945dfc3f9e264c258ef29c45910b7fd8bc/deps/uv/src/win/util.c#L1144